sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/WatchKafkaTopicPartitionDoFn.java

```diff
@@ -107,13 +112,17 @@ public void processElement(
     current.forEach(
         topicPartition -> {
           if (checkStopReadingFn == null || !checkStopReadingFn.apply(topicPartition)) {
+            Counter foundedTopicPartition =
+                Metrics.counter(COUNTER_NAMESPACE, topicPartition.toString());
+            foundedTopicPartition.inc();
             existingTopicPartitions.add(topicPartition);
             outputReceiver.output(
                 KafkaSourceDescriptor.of(topicPartition, null, startReadTime, null));
           }
         });
 
-    timer.set(Instant.now().plus(checkDuration.getMillis()));
+    timer.offset(checkDuration).setRelative();
```

Comment:
Why it requires this change?

Reply:
Because I just realized that `set()` is not allowed for a processing time timer.